tests/e2e/network/service_annotations.go

```diff
@@ -315,3 +355,71 @@ func validateInternalLoadBalancer(c clientset.Interface, ns string, url string)
 	utils.Logf("validation finished")
 	return err
 }
+
+func validateLoadBalancerBackendPools(vmssname string, cs clientset.Interface, labels map[string]string, ns string, ports []v1.ServicePort, resourceGroupName string) {
+
+	serviceName := "validate-lb-mode"
+	lbModeLoadBalancerName := serviceName + string(uuid.NewUUID())
+	// create deployment
+	utils.Logf("Creating deployment " + lbModeLoadBalancerName)
+	deployment := createNginxDeploymentManifest(lbModeLoadBalancerName, labels)
+	_, err := cs.AppsV1().Deployments(ns).Create(deployment)
+	Expect(err).NotTo(HaveOccurred())
+
+	// create annotation for LoadBalancer service
+	annotation := map[string]string{
+		azure.ServiceAnnotationLoadBalancerMode: vmssname,
+	}
+	service := createLoadBalancerServiceManifest(cs, lbModeLoadBalancerName, annotation, labels, ns, ports)
+	_, er := cs.CoreV1().Services(ns).Create(service)
+	Expect(er).NotTo(HaveOccurred())
+	utils.Logf("Successfully created LoadBalancer service " + lbModeLoadBalancerName + " in namespace " + ns)
+
+	//wait and get service's public IP Address
+	By("Waiting for service exposure")
+	pip, err := utils.WaitServiceExposure(cs, ns, serviceName)
+	Expect(err).NotTo(HaveOccurred())
+
+	//Invoking azure network client to get list of public IP Addresses
+	ctx := context.Background()
+	azureTestClient, err := utils.CreateAzureTestClient()
+	Expect(err).NotTo(HaveOccurred())
+	iplist, er := azureTestClient.ListPublicIPs(ctx, resourceGroupName)
+	Expect(er).NotTo(HaveOccurred())
+	var pipID string
+	for _, ip := range iplist {
+		if ip.PublicIPAddressPropertiesFormat != nil && ip.PublicIPAddressPropertiesFormat.IPConfiguration != nil && *ip.PublicIPAddressPropertiesFormat.IPAddress == pip {
+			pipID = *ip.PublicIPAddressPropertiesFormat.IPConfiguration.ID
+			utils.Logf("Public Ip verified")
+		}
+	}
+	//Get Azure loadBalancer Name
+	match := lbNameRE.FindStringSubmatch(pipID)
+	if len(match) != 2 {
+		Skip("Report error")
+	}
+	loadBalancername := match[1]
+	utils.Logf(loadBalancername)
+	//Get backendpools list
+	lbClient, err := utils.GetLoadBalancer()
+
+	backendpools, err := lbClient.ListComplete(ctx, loadBalancername)
+	Expect(err).NotTo(HaveOccurred())
+
+	//check whether vmss name equals the backendpool name
```

Comment:
the following logic is not right. As we discussed early, it should 

* first getting loadBalancer backendPoolID from loadBalancer rules list by comparing rule.FrontendIPConfiguration.ID
* then in another BackendAddressPools  loop, get backend pool by comparing backendPoolID and then 
* finnally, extract the vmss name from pool.BackendIPConfigurations and validate with vmssname.